Neos.ContentRepository.LegacyNodeMigration/Classes/Command/SiteCommandController.php

```diff
@@ -83,71 +72,84 @@ public function migrateLegacyDataCommand(bool $verbose = false, string $config =
         }
         $this->verifyDatabaseConnection($connection);
 
+        $contentRepositoryId = ContentRepositoryId::fromString($contentRepository);
+        $temporaryFilePath = $this->environment->getPathToTemporaryDirectory() . uniqid('Export', true);
+        Files::createDirectoryRecursively($temporaryFilePath);
 
-        $siteRows = $connection->fetchAllAssociativeIndexed('SELECT nodename, name, siteresourcespackagekey FROM neos_neos_domain_model_site');
-        $siteNodeName = $this->output->select('Which site to migrate?', array_map(static fn (array $siteRow) => $siteRow['name'] . ' (' . $siteRow['siteresourcespackagekey'] . ')', $siteRows));
-        assert(is_string($siteNodeName));
-        $siteRow = $siteRows[$siteNodeName];
+        $legacyExportService = $this->contentRepositoryRegistry->buildService(
+            $contentRepositoryId,
+            new LegacyExportServiceFactory(
+                $connection,
+                $resourcesPath,
+                $this->propertyMapper,
+            )
+        );
 
-        $site = $this->siteRepository->findOneByNodeName($siteNodeName);
-        if ($site !== null) {
```

Comment:
cool to have that cleaned up as well ❤️ was not even aware of that mess really ^^